### PR TITLE
Enable stricter TS settings

### LINKED ISSUE #815: Glimmer's TS config should be at least as strict as Ember's
I ran into problems importing certain glimmer packages from TypeScript files in ember.js due to ember.js having stricter TypeScript settings than glimmer-vm.

Ember's current tsconfig:
```json
    "strict": true,
    "noImplicitReturns": true,
    "allowUnreachableCode": false,
    "suppressImplicitAnyIndexErrors": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
```

Glimmer-vm's current tsconfig:
```json
    "noImplicitAny": true,
    "suppressImplicitAnyIndexErrors": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "allowUnreachableCode": false,
    "strictNullChecks": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
```

I have a fix in for `@glimmer/compiler` at #814, but this will become more of a problem as more of ember's codebase switches to TypeScript.

### PR DESCRIPTION
**Pull request by csantero**
Fixes #815

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "moduleResolution": "node",
 
     // Enhance Strictness
+    "strict": true,
```
Comment (t-sauer) [inline tsconfig.json]: I think you can remove the other "strict" compiler options (like `noImplicitAny`) since `stricts` activates them all anyway.

Comment (rwjblue) [reply tsconfig.json]: Nice! I didn’t realize that. Do you know where the docs are for that (just so I can read up on it)?

Comment (csantero) [reply tsconfig.json]: @rwjblue https://www.typescriptlang.org/docs/handbook/compiler-options.html

**Comment by rwjblue** [comment]
I'm going to go ahead and land this (don't want to cause merge conflicts with other changes that are in the works...), I'm happy to work on any cleanups though...

**Comment by rwjblue** [comment]
Thank you again @csantero!

**Pull request merged**